### ISSUE TITLE
segmentation fault when opening mzrollDB

### ISSUE THREAD

**Issue by PMSeitzer** [eugenemel/maven#121]
with `masscalcgui` showing, `compute()` is being called:
```
MainWindow::showSelectAdductsDialog()
SelectAdductsDialog::updateSelectedAdducts()
changeUserAdduct::[M+2H]2+
MainWindow::showSelectAdductsDialog()
SelectAdductsDialog::deselectAll()
SelectAdductsDialog::updateSelectedAdducts()
changeUserAdduct::[M+Cl]-
MassCalcWidget::compute()
Segmentation fault: 11
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
```

**Comment by PMSeitzer** [comment]
Note that when libraries loaded or compound widget is invisible, the bug is avoided.

**Comment by PMSeitzer** [comment]
implemented as of ff494f22a946834cdaa2c356f595ea1ee90ee572
